RELEASING.md

```diff
@@ -141,39 +141,67 @@ Tagging the Release
    ```
 7. Close the release milestone.
 
-Build Artifacts
----------------
+8. Trigger build as described in "Auto releasing using kokoro" at
+   go/grpc/java/releasing.
```

Comment:
Optional, here and elsewhere:

```suggestion
   go/grpc-prod/java/releasing.
```

Reply:
Yeah... but it is also in the wrong location and should get moved to go/grpc-oss/java/releasing. I'm going to leave it as-is for now, since our internal docs are a bit of a rats nest.